sqlglot/parser.py

```diff
@@ -572,93 +539,145 @@ def parse_option(option, token, option_lambda):
             properties=properties,
             temporary=temporary,
             replace=replace,
-            **options,
         )
 
     def _parse_property(self, schema):
-        key = self._parse_var().this
-        self._match(TokenType.EQ)
-
-        if key.upper() == c.PARTITIONED_BY:
-            value = self._parse_schema() or self._parse_bracket(self._parse_field())
-
-            if schema and not isinstance(value, exp.Schema):
-                columns = {v.text("this").upper() for v in value.expressions}
-                partitions = [
-                    expression
-                    for expression in schema.expressions
-                    if expression.this.text("this").upper() in columns
-                ]
-                schema.set(
-                    "expressions",
-                    [e for e in schema.expressions if e not in partitions],
-                )
-                value = self.expression(exp.Schema, expressions=partitions)
-        else:
-            value = self._parse_column()
-
-        return self.expression(
-            exp.Property,
-            this=exp.Literal.string(key),
-            value=value,
-        )
-
-    def _parse_properties(self, schema):
-        properties = []
+        if self._match(TokenType.USING):
```

Comment:
can we move this all to a dictionary with a lambda

TokenType.USING: lambda ... 

that way every dialect can override how it reads property, also then it will be constant time to match

Reply:
Yeah nice improvement.